python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py

```diff
@@ -903,6 +903,8 @@ def get_run_partition_data(self, runs_filter: RunsFilter) -> Sequence[RunPartiti
             for row in rows:
                 if not row["partition"] or row["partition"] in _partition_data_by_partition:
                     continue
+                if DagsterRunStatus[row["status"]] == DagsterRunStatus.CANCELED:
```

Comment:
I think this method is used to display run statuses for backfills which means that canceled backfill runs may not display as expected in the UI. In this case I believe we try to display a run status "bar" for each partition, and since the canceled partitions no longer have runs they'd display as not started?

I think we probably instead edit the partition set runs filter to not select canceling runs here: https://github.com/dagster-io/dagster/blob/429661fed123d71a43309a1f98e3e23145098e03/python_modules/dagster-graphql/dagster_graphql/implementation/fetch_partition_sets.py#L231-L237

Reply:
Ahh it looks like you're right, good catch! I think moving this up into the filter on the original query makes sense, will change this. Glad you noticed this!